Implement chart methods

pfouque opened the pull request:
I needed to use those methods which wasn't available in your module, so i did it.
BTW i'm a bit sorry about quality but this method is the only one to return array of different types and where relations are attached to a chart entity with id 0.
This time i deliver it directly with tests.

browniebroke commented:
Thanks for the PR!

I'll look at it shortly. In the meantime, could you please merge/rebase my changes for coverages from the `master` branch? That would give more visibility on the status.

pfouque commented:
It's done.
I also fixed code coverage by adding more tests on sub resources.
By doing so, i discovered a bug which is now fixed.

I'm not really satisfied with this implementation, especially the specific case in Client init, but as this API is really specific i didn't found another way to implement it in the actual base code without modifying core concept of your module. 

So if you have another way to implement it i'm really curious to see how.

browniebroke commented:
Thanks! Agree with you, but there are tests so we can improve it later if needed.
